### PR TITLE
Fix donation switch network and TokenPicker and Givpower Solana issue

### PR DESCRIPTION
**Pull request by alireza-sharifpour**
#3399

### CODE REVIEW

File: src/config/development.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ const CELO_NETWORK_NUMBER = 44787;
 const CLASSIC_NETWORK_NUMBER = 63;
 const SOLANA_NETWORK: NonEVMChain = {
 	id: 0,
+	networkId: 103,
```
Comment (aminlatifi) [inline src/config/development.tsx:54]: I am suspicious about using `networkId`
What if in the future an EVM chain comes with the same value?!

File: src/components/views/donate/SwitchToAcceptedChain.tsx [hunk 2]
```diff
@@ -10,16 +10,45 @@ import {
 import { INetworkIdWithChain } from './common.types'; // Import the type
 import { useGeneralWallet } from '@/providers/generalWalletProvider';
 import { ChainType } from '@/types/config';
+import config from '@/configuration';
 
 const SwitchToAcceptedChain: FC<{ acceptedChains: INetworkIdWithChain[] }> = ({
```
Comment (aminlatifi) [inline src/components/views/donate/SwitchToAcceptedChain.tsx:15]: INetworkIdWithChain has chainType as well. So, let's rely on that for switching to Solana.

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/config/development.tsx (hunk 1 above)
Comment (alireza-sharifpour) [reply src/config/development.tsx:54]: What is your suggestion? we did this based on the naming of the backend.
I think if some day a chain has added and we decided to support it on Giveth, then we can change the name.
What do you think?

Comment (aminlatifi) [reply src/config/development.tsx:54]: I mean, you don't have to add anything here at this stage with your intended changes. So please don't modify it so far and let it for other PRs

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Comment by alireza-sharifpour** [comment]
Thanks @aminlatifi,
I did the suggested changes and also fixed the bug we had in TokenPicker on this page in this PR.

**Review by aminlatifi — APPROVED** [review]

**Review by RamRamez — APPROVED** [review]
Thanks @alireza-sharifpour

**Pull request merged**